test/state/CMakeLists.txt

```diff
@@ -72,3 +72,16 @@ if(EVMONE_PRECOMPILES_SILKPRE)
         precompiles_silkpre.cpp
     )
 endif()
+
+# This is done after Silkpre because it also tries to find GMP.
+option(EVMONE_PRECOMPILES_GMP "Enable precompiles implementations via the GMP/MPIR library" ON)
```

Comment:
On by default seems to cause a lot of build issues. Maybe use the same strategy as with SILKPRE (OFF by default).